File: Order App/Bidbatl/app/src/main/java/com/android/bidbatl/Utility/ConnectionDetector.java
package com.android.bidbatl.Utility;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class ConnectionDetector {

    private Context context;

    public ConnectionDetector(Context context) {
        this.context = context;
    }

    public final boolean isInternetOn() {

        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        assert cm != null;
        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        if (activeNetwork != null) {
            if (activeNetwork.getType() == ConnectivityManager.TYPE_WIFI) {
                return true;
            } else if (activeNetwork.getType() == ConnectivityManager.TYPE_MOBILE) {
                return true;
            } else if (activeNetwork.getType() == ConnectivityManager.TYPE_MOBILE_DUN) {
                return true;
            }
        } else {

            return false;
        }
        return false;
    }
}

File: DeliveryApp/Bidbatl_Delivery/app/src/main/java/com/bidbatl/dileveryapp/util/ConnectionDetector.kt
package com.bidbatl.dileveryapp.util

import android.content.Context
import android.net.ConnectivityManager
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ConnectionDetector @Inject constructor() {
    fun init(): Boolean {
        return true
    }

    fun isInternetOn(context: Context?): Boolean {
        val cm = (context!!.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager)
        val activeNetwork = cm.activeNetworkInfo
        return if (activeNetwork != null) {
            when (activeNetwork.type) {
                ConnectivityManager.TYPE_WIFI -> {
                    true
                }
                ConnectivityManager.TYPE_MOBILE -> {
                    true
                }
                else -> activeNetwork.type == ConnectivityManager.TYPE_MOBILE_DUN
            }
        } else {
            false
        }
    }
}